src/card/Address.java

```diff
@@ -0,0 +1,164 @@
+		return this.getStreetAddress() + ", " + this.getCity() + ", " + this.getState() + " " + this.getZipCode();
+	}
+	
+	public void setStreetAddress(String streetAddress) {
+		this.addressEntity.setStreetAddress(Encryption.encrypt(streetAddress));
+	}
+	
+	public void setCity(String city) {
+		this.addressEntity.setCity(Encryption.encrypt(city));
+	}
+	
+	public void setState(String state) {
+		this.addressEntity.setState(Encryption.encrypt(state));
+	}
+	
+	public void setZipCode(String zipCode) {
+		this.addressEntity.setZipCode(Encryption.encrypt(zipCode));
+	}
+	
+	protected AddressEntity getAddressEntity() {
+		return this.addressEntity;
+	}
+	
+	public boolean addressExists(ConnectionSource databaseConnection){
+		//checks if an address with this streetAddress, city, state, zipCode exists 
+		try {
+			Dao<AddressEntity, String> addressDao = DaoManager.createDao(databaseConnection, AddressEntity.class);
+			Map<String, Object> queryParams = new HashMap<String, Object>();
+			queryParams.put("street_address", this.addressEntity.getStreetAddress());
+			queryParams.put("city", this.addressEntity.getCity());
+			queryParams.put("state", this.addressEntity.getState());
+			queryParams.put("zip_code", this.addressEntity.getZipCode());
+			
+			List<AddressEntity> returnedAddresses = addressDao.queryForFieldValues(queryParams);
+			if(returnedAddresses.isEmpty()) {
+				return false;
+			}
+			else {
+				return true;
+			}
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		
+		return false; 
+	}
+	
+	private AddressEntity getExistingEntity(ConnectionSource databaseConnection) {
+		try {
+			Dao<AddressEntity, String> addressDao = DaoManager.createDao(databaseConnection, AddressEntity.class);
+			Map<String, Object> queryParams = new HashMap<String, Object>();
+			queryParams.put("street_address", this.addressEntity.getStreetAddress());
+			queryParams.put("city", this.addressEntity.getCity());
+			queryParams.put("state", this.addressEntity.getState());
+			queryParams.put("zip_code", this.addressEntity.getZipCode());
+			
+			List<AddressEntity> returnedAddresses = addressDao.queryForFieldValues(queryParams);
+			if (returnedAddresses.isEmpty()) {
+				return null;
+			}
+			return returnedAddresses.get(0);
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		return null;
+	}
+	
+	/**
+	 * 
+	 * @param databaseConnection connection to the database
+	 * @return true if it updated to an existing address, false if there was not an existing address to update to
+	 */
+	public boolean updateToExistingAddress(ConnectionSource databaseConnection) {
+		if (this.addressExists(databaseConnection)) {
+			AddressEntity updatedEntity = this.getExistingEntity(databaseConnection);
+			this.addressEntity = updatedEntity;
+			return true;
+		}
+		return false;
+	}
+	
+	public boolean addAddress(ConnectionSource databaseConnection) {
```

Comment:
Is there a reason to return a boolean here? If the the insert doesn't work, won't a SQLException just be thrown anyway? What are the other possible return values of create besides 1?

Reply:
it catches the exception so it wouldn't get thrown, this way when we add an address if it fails bc of a SQLException we can at least alert the user that it failed